<?xml version="1.0"?>
<launch>
  <!-- The TSV (tab-separated-values) file to read the trajectory data from.
  The first line must be the TSV header and the columns should be: timestamp in
  nanoseconds, tWB_x, tWB_y, tWB_z, qWB_x, qWB_y, qWB_z and qWB_w. The position
  must be in meters. -->
  <arg name="tsv_file" default="$(find srl_mpc_examples)/data/depot_trajectory_0.tsv"/>

  <!-- Whether to flush the MPC queue to execute the trajectory immediately. -->
  <arg name="flush_queue" default="false"/>

  <!-- Read every n states instead of all of them. -->
  <arg name="read_every_n" default="1"/>

  <node pkg="srl_mpc_examples" type="publish_trajectory.py" name="publish_trajectory">
    <param name="tsv_file" type="string" value="$(arg tsv_file)"/>
    <param name="flush_queue" type="bool" value="$(arg flush_queue)"/>
    <param name="read_every_n" type="int" value="$(arg read_every_n)"/>
    <remap from="~trajectory" to="/autopilot/trajectory"/>
  </node>
</launch>
